testplan/runnable/interactive/base.py

```diff
@@ -813,6 +813,33 @@ def _stop_environment(self, test_uid, stop_generator):
         else:
             self._set_env_status(test_uid, entity.ResourceStatus.STOPPED)
 
+    @contextlib.contextmanager
+    def _auto_start_stop_environment(self, test_uid, runner_uid):
+        """Context manager for automatic environment control."""
+        test = self.test(test_uid, runner_uid=runner_uid)
+        resources_started = False
+
+        if not test.resources.all_status(entity.ResourceStatus.STARTED):
+            if not test.resources.all_status(
+                entity.ResourceStatus.STOPPED
+            ) and not test.resources.all_status(entity.ResourceStatus.NONE):
+                # State requires to reset all.
+                self.stop_test_resources(
+                    test_uid,
+                    runner_uid=runner_uid,
+                )
+            self.start_test_resources(
+                test_uid, runner_uid=runner_uid,
+            )
+            resources_started = True
+
+        yield
+
+        if resources_started:
```

Comment:
I still don't think we should stop the resources after running individual tests.